web-api/src/personalization/rerank.rs

```diff
@@ -163,3 +163,190 @@ pub fn bench_rerank<S>(
         time,
     );
 }
+
+#[cfg(test)]
+mod tests {
+    use std::{collections::HashSet, time::Duration};
+
+    use xayn_ai_bert::Embedding1;
+    use xayn_ai_coi::{CoiConfig, CoiId, CoiStats, PositiveCoi};
+    use xayn_test_utils::assert_approx_eq;
+
+    use super::*;
+
+    fn mock_documents(n: usize) -> Vec<PersonalizedDocument> {
+        (0..n)
+            .map(|i| {
+                let id = i.to_string().try_into().unwrap();
+
+                let mut embedding = vec![1.0; n];
+                embedding[i] = 10.0;
+                let embedding = Embedding1::from(embedding).normalize().unwrap();
+
+                let tags = if i % 2 == 0 {
+                    vec!["general".try_into().unwrap()]
+                } else {
+                    vec![
+                        "general".try_into().unwrap(),
+                        "specific".try_into().unwrap(),
+                    ]
+                };
+
+                PersonalizedDocument {
+                    id,
+                    score: 1.0,
+                    embedding,
+                    properties: DocumentProperties::default(),
+                    tags,
+                }
+            })
+            .collect()
+    }
+
+    fn mock_coi(i: usize, n: usize, time: DateTime<Utc>) -> PositiveCoi {
+        let id = CoiId::new();
+
+        let mut point = vec![0.0; n];
+        point[i] = 1.0;
+        let point = Embedding1::from(point).normalize().unwrap();
+
+        let stats = CoiStats {
+            view_count: i + 1,
+            view_time: Duration::ZERO,
+            last_view: time,
+        };
+
+        PositiveCoi { id, point, stats }
+    }
+
+    fn sort_ids(scores: HashMap<DocumentId, f32>) -> Vec<String> {
```

Comment:
:art:  I noticed that when reading the tests I was thinking "if I sort the id don't I always get the same order?"

Maybe renaming it  it to `ids_sorted_by_score` or `ids_by_score` or similar would be a bit better :thinking: